### DISCUSSION TITLE
Puppeteer with nitro on serverless env

### DISCUSSION THREAD

**Discussion by juane1000** [nitrojs/nitro#2414] [Q&A]
When using Puppeteer there's an issue where chrome isn't found. My understanding is that a prod build will bundle all dependencies. When I deploy to firebase cloud functions there's no indication that 'npm  ci' is being run in the logs so chrome is never downloaded for Puppeteer to use.

Not sure how to proceed. Is it possible to exclude some dependencies from the build bundle so that they're installed on the remote environment? There is a puppeteer npx command to manually download chrome but no obvious way to execute it with nitro + firebase deploy.

My use case is having a server route do a simple scrape of some web page using Puppeteer. It works locally on the dev server but this issue with bundlers is documented on Puppeteer's own site with no workable solution for nitro mentioned.

**Comment by stafyniaksacha** [comment]
Using serverless hosting, this will be near impossible because it will rely on binaries. You can take a look at cloudflare browser rendering https://developers.cloudflare.com/browser-rendering/, but I don't think there will be an equivalent in firebase.

**Reply by juane1000** [reply]
Reading through the cloudflare, it looks promising. Thank you very much!
